### PR TITLE
[controller] fix issues in #30260

### PR DESCRIPTION
**Pull request by erjiaqing**

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -1177,18 +1177,25 @@ void DeviceCommissioner::OnICDManagementRegisterClientResponse(
...
+        return;
     }
-    else
+
+    if (commissioner->mDeviceBeingCommissioned == nullptr)
+    {
+        return;
+    }
+
+    if (commissioner->mPairingDelegate != nullptr)
```
Comment (bzbarsky-apple) [inline src/controller/CHIPDeviceController.cpp:1192]: So if there is no pairing delegate... should we not still call CommissioningStageComplete()?

Comment (yunhanw-google) [reply src/controller/CHIPDeviceController.cpp:1192]: yes, we still need call CommissioningStageComplete(). just update, thanks

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: src/controller/CHIPDeviceController.cpp [hunk 2]
```diff
@@ -1177,19 +1177,25 @@ void DeviceCommissioner::OnICDManagementRegisterClientResponse(
     void * context, const app::Clusters::IcdManagement::Commands::RegisterClientResponse::DecodableType & data)
 {
     DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
+    VerifyOrReturn(commissioner != nullptr, ChipLogProgress(Controller, "Command response callback with null context. Ignoring"));
 
-    CommissioningDelegate::CommissioningReport report;
-    auto pairingDelegate         = commissioner->GetPairingDelegate();
-    auto deviceBeingCommissioned = commissioner->mDeviceBeingCommissioned;
-    if (pairingDelegate != nullptr && deviceBeingCommissioned != nullptr)
+    if (commissioner->mCommissioningStage != CommissioningStage::kICDRegistration)
```
Comment (mkardous-silabs) [inline src/controller/CHIPDeviceController.cpp:1182]: Suggestion if this is touched later 
`VerifyOrReturn(commissioner->mCommissioningStage == CommissioningStage::kICDRegistration);`

File: src/controller/CHIPDeviceController.cpp [hunk 3]
```diff
@@ -1177,19 +1177,25 @@ void DeviceCommissioner::OnICDManagementRegisterClientResponse(
...
     {
-        pairingDelegate->OnICDRegistrationComplete(deviceBeingCommissioned->GetDeviceId(), data.ICDCounter);
-        commissioner->CommissioningStageComplete(CHIP_NO_ERROR, report);
+        return;
     }
-    else
+
+    if (commissioner->mDeviceBeingCommissioned == nullptr)
```
Comment (mkardous-silabs) [inline src/controller/CHIPDeviceController.cpp:1187]: Suggestion if this is touched later 
`VerifyOrReturn(commissioner->mDeviceBeingCommissioned != nullptr);`

**Pull request merged**